src/services/import.service.ts

```diff
@@ -0,0 +1,229 @@
+import { ImportService as ImportServiceAbstraction } from '../abstractions/import.service';
+
+import { ImportResult } from '../models/domain/importResult';
+
+import { CipherRequest } from '../models/request/cipherRequest';
+import { FolderRequest } from '../models/request/folderRequest';
+import { ImportCiphersRequest } from '../models/request/importCiphersRequest';
+import { KvpRequest } from '../models/request/kvpRequest';
+
+import { CipherView } from '../models/view/cipherView';
+
+import { AscendoCsvImporter } from '../importers/ascendoCsvImporter';
+import { AviraCsvImporter } from '../importers/aviraCsvImporter';
+import { BitwardenCsvImporter } from '../importers/bitwardenCsvImporter';
+import { BlurCsvImporter } from '../importers/blurCsvImporter';
+import { ChromeCsvImporter } from '../importers/chromeCsvImporter';
+import { ClipperzHtmlImporter } from '../importers/clipperzHtmlImporter';
+import { DashlaneCsvImporter } from '../importers/dashlaneCsvImporter';
+import { EnpassCsvImporter } from '../importers/enpassCsvImporter';
+import { FirefoxCsvImporter } from '../importers/firefoxCsvImporter';
+import { GnomeJsonImporter } from '../importers/gnomeJsonImporter';
+import { Importer } from '../importers/importer';
+import { KeePass2XmlImporter } from '../importers/keepass2XmlImporter';
+import { KeePassXCsvImporter } from '../importers/keepassxCsvImporter';
+import { KeeperCsvImporter } from '../importers/keeperCsvImporter';
+import { LastPassCsvImporter } from '../importers/lastpassCsvImporter';
+import { MeldiumCsvImporter } from '../importers/meldiumCsvImporter';
+import { MSecureCsvImporter } from '../importers/msecureCsvImporter';
+import { OnePassword1PifImporter } from '../importers/onepassword1PifImporter';
+import { OnePasswordWinCsvImporter } from '../importers/onepasswordWinCsvImporter';
+import { PadlockCsvImporter } from '../importers/padlockCsvImporter';
+import { PassKeepCsvImporter } from '../importers/passkeepCsvImporter';
+import { PasswordAgentCsvImporter } from '../importers/passwordAgentCsvImporter';
+import { PasswordBossJsonImporter } from '../importers/passwordBossJsonImporter';
+import { PasswordDragonXmlImporter } from '../importers/passwordDragonXmlImporter';
+import { PasswordSafeXmlImporter } from '../importers/passwordSafeXmlImporter';
+import { RoboFormCsvImporter } from '../importers/roboformCsvImporter';
+import { SafeInCloudXmlImporter } from '../importers/safeInCloudXmlImporter';
+import { SaferPassCsvImporter } from '../importers/saferpassCsvImport';
+import { SplashIdCsvImporter } from '../importers/splashIdCsvImporter';
+import { StickyPasswordXmlImporter } from '../importers/stickyPasswordXmlImporter';
+import { TrueKeyCsvImporter } from '../importers/truekeyCsvImporter';
+import { UpmCsvImporter } from '../importers/upmCsvImporter';
+import { ZohoVaultCsvImporter } from '../importers/zohoVaultCsvImporter';
+
+export interface ISubmitResult {
```

Comment:
Is this really needed? Could you just return a string for success and reject with message on failure? Not sure what `shouldReport` is for though.

Reply:
I wasn't sure what to do here, actually.  In bitwarden/web angularlytics is called inline for certain scenarios.  In order to preserve that behavior, I really needed to pass an object back.

If you have a suggestion for how to more appropriately do this (or if you'd be ok with `any` here) let me know.  I'm pretty new to Typescript.

Thanks for the very quick review.

Reply:
I would consider maybe just leaving some of those "precheck" errors (should report = false) in each client application rather than keeping them here.